src/AutoMapper.php

```diff
@@ -48,6 +51,22 @@ public static function initialize(callable $configurator): AutoMapperInterface
         return $mapper;
     }
 
+    private function push($key, $value, &$context)
+    {
+        if (!array_key_exists($key, $context)) {
+            $stack = [];
```

Comment:
How do you feel about using an actual stack (e.g. SplStack) instead of an array? I think using SplStack makes more sense for what we're doing here (pushing/popping). But for the end user an array would be more convenient when they are doing something with the context (e.g. all the `array_` functions are available to them etc.). So I guess we'd best stick to an array :)

Reply:
I think we should stick an array too. Maybe I could extract push/pop private methods into a utility class though (in the middlareware pull requests, those are duplicated)